src/net/sourceforge/kolmafia/swingui/widget/AutoFilterTextField.java

```diff
@@ -223,73 +223,66 @@ public static final int getResultQuantity(final Object element) {
   }
 
   public void update() {
-    try {
-      AutoFilterTextField.this.qtyChecked = false;
-      AutoFilterTextField.this.asChecked = false;
-      AutoFilterTextField.this.notChecked = false;
-      AutoFilterTextField.this.text = AutoFilterTextField.this.getText().toLowerCase();
-
-      if (AutoFilterTextField.this.text != null
-          && AutoFilterTextField.this.text.length() > 1
-          && AutoFilterTextField.this.text.charAt(0) == "-".charAt(0)) {
-        AutoFilterTextField.this.notChecked = true;
-        AutoFilterTextField.this.text = AutoFilterTextField.this.text.substring(1).trim();
-      }
+    AutoFilterTextField.this.qtyChecked = false;
+    AutoFilterTextField.this.asChecked = false;
+    AutoFilterTextField.this.notChecked = false;
+    AutoFilterTextField.this.text = AutoFilterTextField.this.getText().toLowerCase();
+
+    if (AutoFilterTextField.this.text != null
```

Comment:
IntelliJ says AutoFilterTextField.this.text != null is always true.

But if I am reading right AutoFilterTextField.this.getText() can return null.  So it is either always true or we have previously had some kind of error caused by .toLowerCase() on a null value, right?

Reply:
If getText() returns null, then toLowerCase() will result in a NPE, yeah.